### PR TITLE
fix e2e ingress test, when we enable the ingress webhook

### PR DESCRIPTION
**Pull request by xlgao-zju**
Signed-off-by: Xianglin Gao <xianglin.gxl@alibaba-inc.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
When we run the e2e test against a k8s cluster installed with a ingress webhook(https://github.com/kubernetes/ingress-nginx/blob/5f1a37a624ca38e8cccc87cb7a36d7dbcbe70b01/deploy/static/provider/baremetal/deploy.yaml#L586), we will get this failure. 

```
[sig-network] Ingress API
  should support creating Ingress API operations [Conformance]
  /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:629
[BeforeEach] [sig-network] Ingress API
  /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:174
STEP: Creating a kubernetes client
Mar 22 06:54:11.386: INFO: >>> kubeConfig: /tmp/kubeconfig-926267848
STEP: Building a namespace api object, basename ingress
STEP: Binding the e2e-test-privileged-psp PodSecurityPolicy to the default service account in ingress-2026
STEP: Waiting for a default service account to be provisioned in namespace
[It] should support creating Ingress API operations [Conformance]
  /workspace/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:629
STEP: getting /apis
STEP: getting /apis/networking.k8s.io
STEP: getting /apis/networking.k8s.iov1
STEP: creating
Mar 22 06:54:11.558: FAIL: Unexpected error:
    <*errors.StatusError | 0xc000bf75e0>: {
        ErrStatus: {
            TypeMeta: {Kind: "", APIVersion: ""},
            ListMeta: {
                SelfLink: "",
                ResourceVersion: "",
                Continue: "",
                RemainingItemCount: nil,
            },
            Status: "Failure",
            Message: "admission webhook \"validate.nginx.ingress.kubernetes.io\" denied the request: host \"foo.bar.com\" and path \"/\" is already defined in ingress ingress-2026/e2e-example-ingfddnc",
            Reason: "BadRequest",
            Details: nil,
            Code: 400,
        },
    }
    admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "foo.bar.com" and path "/" is already defined in ingress ingress-2026/e2e-example-ingfddnc
occurred
```
Look like we get this failure, since we create the same ingress too many times.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e/network/ingress.go [hunk 1]
```diff
@@ -1015,10 +1015,6 @@ var _ = common.SIGDescribe("Ingress API", func() {
 
 		// Ingress resource create/read/update/watch verbs
 		ginkgo.By("creating")
-		_, err := ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
-		framework.ExpectNoError(err)
-		_, err = ingClient.Create(context.TODO(), ingTemplate, metav1.CreateOptions{})
-		framework.ExpectNoError(err)
```
Comment (aojea) [inline test/e2e/network/ingress.go:1021]: this is part of the test, you can't remove it

**Comment by aojea** [comment]
> Look like we get this failure, since we create the same ingress too many times.

I didn't dig too much into the test but it seems the test is working as expected, if you remove those lines the test fails with
```
/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/network/common/framework.go:23
  should support creating Ingress API operations [Conformance] [It]
  /home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:630
  Mar 22 11:53:20.222: filtered list should have 3 items
      
  Expected
      <int>: 1
  to equal
      <int>: 3
```

should not be better to skip the test if you are using a custom ingress webhook?

**Comment by xlgao-zju** [comment]
@aojea Looks like the ingress webhook(https://github.com/kubernetes/ingress-nginx/blob/master/deploy/static/provider/baremetal/deploy.yaml#L586) is the recommended webhook by us, so should we support the webhook in the e2e test?

And maybe we should create three ingresses with different `Host` and `Path`? Maybe it will be more meaningful?

**Comment by aojea** [comment]
> And maybe we should create three ingresses with different `Host` and `Path`? Maybe it will be more meaningful?

So, the test create 3 ingresses resources using a template with the same rule.
I really don't have the knowledge to understand if this should be allowed or not, let me ping people with more knowledge in the area
@robscott @cmluciano @aledbf can you please chime in and explain which behaviour is correct?
 is the test or the webhook wrong?

**Comment by aledbf** [comment]
> admission webhook "validate.nginx.ingress.kubernetes.io" denied the request: host "foo.bar.com" and path "/" is already defined in ingress ingress-2026/e2e-example-ingfddnc

This is working as expected. Defining more than one Ingress for the same hostname and path is not allowed.
(basically without this validation which one should be used?)

> is the test or the webhook wrong?

There are no k8s validations for overlaps in ingress definitions. This is up to each ingress controller implementation.
The test has no knowledge about the ingress controller being used and ingress-nginx does not allow duplications.

I think the easiest fix is to remove the webhook validation in ingress-nginx.

>   Mar 22 11:53:20.222: filtered list should have 3 items

The test is only checking the number of definitions. No traffic is being sent or checked.

@xlgao-zju keep in mind this test creates a GCE controller [here](https://github.com/kubernetes/kubernetes/blob/462c119bba1fd9d357904d35e99a3c21b614d393/test/e2e/network/ingress.go#L96-L105)

**Comment by fusida** [comment]
i don't think it's a good idea to delete all of it.

**Comment by fusida** [comment]
it don't think it is a good idea to delete all of it.
run three times and not just use special .metadata.nam but also use special 'host + port'

**Comment by fusida** [comment]
it don't think it is a good idea to uninstall ingress-nginx webhook or delete e2e-test/ingress.go code in line 1020.
run three times and not just use unique .metadata.nam but also use unique 'host + path'.

**Pull request closed without merging**